d2ir/compile.go

```diff
@@ -210,18 +210,15 @@ func (c *compiler) overlayVars(base, overlay *Map) {
 	if vars == nil {
 		return
 	}
+	vars = vars.Copy(base).(*Field)
```

Comment:
wouldn't this test have caught that? https://github.com/terrastruct/d2/pull/1473/commits/301bc8ddfbd789549af58e16add68a80d9ce84cb

Reply:
Thing is you're passing them into the function flipped too lol

Reply:
Looks like `overlayVars` isn't required. The code being incorrect like I described isn't being noticed because even if you comment it out or flip the base/overlay arguments to it, the tests still pass though the testData changes slightly.

I think the varStack is being kept across boards and so you don't need to overlay vars for the parent board's vars to take into effect if the child board doesn't have its own.

Reply:
It doesn't have any effect because the new vars field is appended to the end of the board whereas it should be prepended if we wanted it to take effect. But we don't since things work anyway.

Reply:
ohhhh right.